libcalico-go/lib/metricsserver/metrics_server_test.go

```diff
@@ -0,0 +1,367 @@
+			certFile:       certFile,
+			keyFile:        keyFile,
+			minTLSVersion:  "TLS13",
+			clientAuthType: "NoClientCert",
+			caFile:         "",
+			clientHasCA:    false,
+			expectedStatus: 200,
+			expectError:    false,
+		},
+		{
+			name:                 "Valid TLS: NoClientCert, TLS13, Rotate Certificates",
+			certFile:             certFile,
+			keyFile:              keyFile,
+			minTLSVersion:        "TLS13",
+			clientAuthType:       "NoClientCert",
+			caFile:               "",
+			clientHasCA:          false,
+			expectedStatus:       200,
+			expectError:          false,
+			rotateCertificates:   true,
+			rotateExpectError:    false,
+			rotateExpectedStatus: 200,
+		},
+		{
+			name:           "Valid TLS: NoClientCert, TLS12",
+			certFile:       certFile,
+			keyFile:        keyFile,
+			minTLSVersion:  "TLS12",
+			clientAuthType: "NoClientCert",
+			caFile:         "",
+			clientHasCA:    false,
+			expectedStatus: 200,
+			expectError:    false,
+		},
+		{
+			name:           "Valid TLS: RequireAndVerifyClientCert, TLS13",
+			certFile:       certFile,
+			keyFile:        keyFile,
+			minTLSVersion:  "TLS13",
+			clientAuthType: "RequireAndVerifyClientCert",
+			caFile:         caFile,
+			clientHasCA:    true,
+			expectedStatus: 200,
+			expectError:    false,
+		},
+		{
+			name:                 "Valid TLS: RequireAndVerifyClientCert, TLS13, Rotate Certificates",
+			certFile:             certFile,
+			keyFile:              keyFile,
+			minTLSVersion:        "TLS13",
+			clientAuthType:       "RequireAndVerifyClientCert",
+			caFile:               caFile,
+			clientHasCA:          true,
+			expectedStatus:       200,
+			expectError:          false,
+			rotateCertificates:   true,
+			rotateExpectError:    false,
+			rotateExpectedStatus: 200,
+		},
+		{
+			name:           "Valid TLS: RequireAndVerifyClientCert, TLS12",
+			certFile:       certFile,
+			keyFile:        keyFile,
+			minTLSVersion:  "TLS12",
+			clientAuthType: "RequireAndVerifyClientCert",
+			caFile:         caFile,
+			clientHasCA:    true,
+			expectedStatus: 200,
+			expectError:    false,
+		},
+		{
+			name:           "Valid TLS: RequireAndVerifyClientCert, TLS13, Client missing valid CA",
+			certFile:       certFile,
+			keyFile:        keyFile,
+			minTLSVersion:  "TLS12",
+			clientAuthType: "RequireAndVerifyClientCert",
+			caFile:         caFile,
+			clientHasCA:    false,
+			expectError:    true,
+		},
+	} {
+		t.Run(tt.name, func(t *testing.T) {
+			RegisterTestingT(t)
+
+			// Dynamically find an open port
+			listener, err := net.Listen("tcp", ":0")
+			if err != nil {
+				t.Fatalf("Failed to find an open port %v", err)
+			}
+			port := listener.Addr().(*net.TCPAddr).Port
+			listener.Close()
+
+			// Start the HTTPS metrics server in a goroutine
+			go ServePrometheusMetricsHTTPS(host, port, tt.certFile, tt.keyFile, tt.minTLSVersion, tt.clientAuthType, tt.caFile)
+
+			// Wait for the server to start
+			time.Sleep(1 * time.Second)
```

Comment:
Instead of a sleep, can we use a retry on the request? That's more resilient to timing issues, and also speeds up the tests in the mainline case.